cmd/filter-proxy/main.go

```diff
@@ -363,28 +386,59 @@ func authorizeRequestWithService(config *config.Config, backend config.Backend,
 		"ip":         utils.ReadUserIP(r),
 	}
 
+	isTransactionSet := false
+
 	if backend.Type == "OWS" {
 		queryParams := utils.QueryParamsToLower(r.URL.Query())
-		authorizationBody["service"] = queryParams.Get("service")
-		authorizationBody["request"] = queryParams.Get("request")
+		var transaction wfs.Transaction
+
+		requestParam := queryParams.Get("request")
+		serviceParam := queryParams.Get("service")
+
+		if len(body) > 0 && requestParam != "" {
+			log.Printf("Invalid: both XML and query param are set")
+			return http.StatusBadRequest, nil, false
+		}
+
+		err := xml.Unmarshal(body, &transaction)
+		transactionSet := transaction.XMLName.Local != ""
+		isTransactionSet = transactionSet
+
+		if len(body) > 0 && err != nil {
+			log.Printf("Body present but could not unmarshal into WFS.Transaction: %v", err)
+			return http.StatusBadRequest, nil, false
+		}
+
+		authorizationBody["service"] = serviceParam
+		authorizationBody["request"] = requestParam
 
 		if authorizationBody["service"] == "WMS" {
 			authorizationBody["resource"] = queryParams.Get("layers") + queryParams.Get("layer")
 			authorizationBody["params"] = map[string]interface{}{
-				"service":    queryParams.Get("service"),
-				"request":    queryParams.Get("request"),
+				"service":    serviceParam,
+				"request":    requestParam,
 				"cql_filter": queryParams.Get("cql_filter"),
 			}
 		} else if authorizationBody["service"] == "WFS" {
 			authorizationBody["resource"] = queryParams.Get("typename") + queryParams.Get("typenames")
 			authorizationBody["params"] = map[string]interface{}{
-				"service":    queryParams.Get("service"),
-				"request":    queryParams.Get("request"),
+				"service":    serviceParam,
+				"request":    requestParam,
 				"cql_filter": queryParams.Get("cql_filter"),
 			}
+		} else if transactionSet {
```

Comment:
I think this needs some refactoring, as a Transaction is a WFS request. So it should be handled inside the 

```
} else if authorizationBody["service"] == "WFS" {
```

block.